### PR TITLE
Fix/cors error pubdev example

### PR DESCRIPTION
**Pull request by PapiHack**
# Fix(#2218): Attempt to fix CORS error on the pub.dev example on web platform

Hello @rrousselGit, hope you're doing well.
This PR is an attempt to fix CORS error on the pub.dev example on web platform.
For short, i use an open source cors proxy server in order to bypass the issue on web thanks to a custom interceptor.
Its working for now but im not sure if it is the best way to handle this :)

### CODE REVIEW

File: examples/pub/lib/pub_repository.dart [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
+    if (kIsWeb) {
+      dio.interceptors.add(PubProxyInterceptor());
+    }
+  }
+
   static const _scheme = 'https';
   static const _host = 'pub.dartlang.org';
-
-  final dio = Dio();
+  late final Dio dio;
```
Comment (rrousselGit) [inline examples/pub/lib/pub_repository.dart:19]: Could you remove the "late" by making a function for the `Dio()` creation?

Comment (rrousselGit) [inline examples/pub/lib/pub_repository.dart:19]:
```suggestion

  late final Dio dio;
```

File: examples/pub/lib/pub_repository.dart [hunk 2]
```diff
@@ -258,3 +265,15 @@ class PubSearchResponse with _$PubSearchResponse {
   factory PubSearchResponse.fromJson(Map<String, Object?> json) =>
       _$PubSearchResponseFromJson(json);
 }
+
+class PubProxyInterceptor extends Interceptor {
```
Comment (rrousselGit) [inline examples/pub/lib/pub_repository.dart:269]: A nice comment to explain why we do this would be great :)

**Comment by rrousselGit** [comment]
Looks epic! Thanks for this!

There are minor points, but otherwise it's great :)

File: examples/pub/lib/pub_repository.dart (hunk 1 above)
Comment (PapiHack) [reply examples/pub/lib/pub_repository.dart:20]: Got it.

File: examples/pub/lib/pub_repository.dart (hunk 2 above)
Comment (PapiHack) [reply examples/pub/lib/pub_repository.dart:269]: Right, no problem.

**Comment by rrousselGit** [comment]
Could you pull changes from the master branch to fix the CI?

**Comment by rrousselGit** [comment]
Looking good~ Thanks for your work! :)

**Comment by PapiHack** [comment]
> Looking good~ Thanks for your work! :)

Great, you're welcome 💯

**Pull request merged**